cli/slsa-verifier/main.go

```diff
@@ -34,6 +34,7 @@ For more information on SLSA, visit https://slsa.dev`,
 	c.AddCommand(verifyImageCmd())
 	// We print our own errors and usage in the check function.
 	c.SilenceErrors = true
+	c.SilenceUsage = true
```

Comment:
Let me see: so if a flag is parsed incorrectly with this change it'll currently print
```
$ go run ./cli/slsa-verifier/ verify-artifact --provenance testdata/go/v1.2.0/binary-linux-amd64-workflow_dispatch.intoto.jsonl --source-uri github.com/slsa-framework/example-package testdata/go/v1.2.0/binary-linux-amd64-workflow_dispatch
unknown flag: --provenance
```

The problem is, that it can't distinguish between error types from flag parsing or generally from RunE. You'll see the usage string only with `--help`. 

This seems like others must have run into this issue. Looking into it.

Reply:
Update: There's no way to distinguish a flag parsing error from a RunE error, unless we handle them separately: https://github.com/spf13/cobra/issues/340

I think the better thing to do is to switch from RunE to Run and use our own custom error handler. Will update today

Reply:
SG, i think this is what I've done in the past for this reason but my memory was fuzzy.